src/cuda/poisson_fft.f90

```diff
@@ -0,0 +1,239 @@
+module m_cuda_poisson_fft
+   use cudafor
+   use cufft
+
+   use m_allocator, only: field_t
+   use m_common, only: dp
+   use m_poisson_fft, only: poisson_fft_t
+   use m_tdsops, only: dirps_t
+
+   use m_cuda_allocator, only: cuda_field_t
+   use m_cuda_common, only: SZ
+   use m_cuda_complex, only: reorder_cmplx_x2y_T, reorder_cmplx_y2x_T, &
+                             reorder_cmplx_y2z_T, reorder_cmplx_z2y_T, &
+                             processfftdiv
+
+   implicit none
+
```

Comment:
Might want to make the module `private` - I think, e.g., `init()` is declared in several places and (potentially) is not desired to be visible outside the module.